main.cpp

```diff
@@ -56,46 +56,83 @@ Purpose:  This project will show you the difference between member functions and
     {
-        if( a->value < b->value ) return a;
-        if( a->value > b->value ) return b;
-        return nullptr;
+        if( a!= nullptr && b != nullptr)
+        {
+            if( a->value < b->value ) return a;
+            if( a->value > b->value ) return b;
+        }
+        
+        //return nullptr; //error: cannot deduce return type 'auto *' from returned value of type 'nullptr_t'
+        T* nullpointer = nullptr; //weird, but works
+        return nullpointer;
     }
 };
 
 struct U
 {
-    float <#name1#> { 0 }, <#name2#> { 0 };
-    <#returnType#> <#memberFunction#>(<#type name#>* <#updatedValue#>)      //12
+    float uFlt1 { 0 }, uFlt2 { 0 };
+    float updatingMmbFunc(float* nuMmbFlt)      //12
     {
-        
+        if(nuMmbFlt == nullptr)
+        {
+            std::cout << "Please check your pointer. Returning 0." << std::endl;
+            return 0.f;
+        }
+
+        std::cout << "U's uFlt1 value: " << this->uFlt1 << std::endl;
+        this->uFlt1 = *nuMmbFlt;
+        std::cout << "U's uFlt1 updated value: " << this->uFlt1 << std::endl;
+        while( std::abs(this->uFlt2 - this->uFlt1) > 0.001f )
+        {
+            /*
+             write something that makes the distance between that-><#name2#> and that-><#name1#> get smaller
+             */
+            this->uFlt2 += 0.0005f;
+        }
+        std::cout << "U's uFlt2 updated value: " << this->uFlt2 << std::endl;
+        return this->uFlt2 * this->uFlt1;
     }
 };
 
-struct <#structname2#>
+struct UpdateR
 {
-    static <#returntype#> <#staticFunctionA#>(U* that, <#type name#>* <#updatedValue#> )        //10
+    static float updatingFunc(U* that, float* nuFlt )        //10
     {
-        std::cout << "U's <#name1#> value: " << that-><#name1#> << std::endl;
-        that-><#name1#> = <#updatedValue#>;
-        std::cout << "U's <#name1#> updated value: " << that-><#name1#> << std::endl;
-        while( std::abs(that-><#name2#> - that-><#name1#>) > 0.001f )
+        if(that != nullptr && nuFlt != nullptr)
         {
-            /*
-             write something that makes the distance between that-><#name2#> and that-><#name1#> get smaller
-             */
-            that-><#name2#> += ;
+            std::cout << "U's uFlt1 value: " << that->uFlt1 << std::endl;
+            that->uFlt1 = *nuFlt;
+            std::cout << "U's uFlt1 updated value: " << that->uFlt1 << std::endl;
+            while( std::abs(that->uFlt2 - that->uFlt1) > 0.001f )
+            {
+                /*
+                write something that makes the distance between that-><#name2#> and that-><#name1#> get smaller
+                */
+                that->uFlt2 += 0.0005f;
+            }
+            std::cout << "U's uFlt2 updated value: " << that->uFlt2 << std::endl;
+            return that->uFlt2 * that->uFlt1;
+        }
+        else
+        {
+            std::cout << "Please check your pointers. Returning 0." << std::endl;
+            return 0.f;
```

Comment:
If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
You do not need the `else` block.  Here's why:
`return` statements are the last thing that **ever happens** in a function.
When the `if()` expression is evaluated as `true` and the block is entered, the `return aValue;` is the last line of code to be executed before the function exits.
but when the `if()` expression evaluates as `false` and the block is **NOT** entered, the next thing that happens is `return anotherValue;`
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
that means you do **NOT** need the `else` block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    
    return anotherValue;
}
```
- [ ] revise